AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/SignInEventData.swift

```diff
@@ -13,15 +13,19 @@ struct SignInEventData {
 
     let clientMetadata: [String: String]
 
+    let deviceMetadata: DeviceMetadata
```

Comment:
Device meta data is not part of signIn event right? I guess we should move this logic to fetch the device data inside the action where it is required

Reply:
Mmmm... I felt like it is part of sign in.. As part of sign in event, we say this is the device metadata that is trying to sign in .. 
If its not part of sign in event, didn't find anything else, which made logical sense to have device data passed into.

Reply:
Fixed... Moved it `InitializeSignInFlow` and added a separate associated value for device metadata